### PR TITLE
Add setBaudRate and setDebugStream to Modem class

### PR DESCRIPTION
**Pull request by dzindra**
Sometimes it is useful to slow down modem baud rate and send debug info somewhere else. Additions do not change default behaviour (baud rate 921600 and Serial as debug output)

### CODE REVIEW

File: src/Modem.h [hunk 1]
```diff
@@ -83,6 +86,8 @@ class ModemClass {
   #define MAX_URC_HANDLERS 10 // 7 sockets + GPRS + GSMLocation + GSMVoiceCall
   static bool _debug;
   static ModemUrcHandler* _urcHandlers[MAX_URC_HANDLERS];
+
+  Stream* _debugStream;
```
Comment (sandeepmistry) [inline src/Modem.h]: I think this can be of type [Print](https://github.com/arduino/ArduinoCore-samd/blob/master/cores/arduino/Print.h) instead of Stream.

Comment (dzindra) [reply src/Modem.h]: Good catch, should I rename method to setDebugPrint?

Comment (dzindra) [reply src/Modem.h]: Also I will probably add `static` keyword since `_debug` flag is also static

**Comment by sandeepmistry** [comment]
Hi @dzindra,

After discussing with @Rocketct outside of Github, we're thinking of overloading the `debug` API:
```
void debug(Print& print);
```

I'll push some commits to this branch before we merge.

**Comment by sandeepmistry** [comment]
@dzindra thanks!

**Pull request merged**